No need to check for -lstdc++.
For one, we're on clang.
For two, the define it sets is never used in the PolyML code.

Index: configure
--- configure.orig
+++ configure
@@ -19414,46 +19414,6 @@ rm -rf conftest*
 fi
 
 
-# Checks for libraries.
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for main in -lstdc++" >&5
-printf %s "checking for main in -lstdc++... " >&6; }
-if test ${ac_cv_lib_stdcpp_main+y}
-then :
-  printf %s "(cached) " >&6
-else $as_nop
-  ac_check_lib_save_LIBS=$LIBS
-LIBS="-lstdc++  $LIBS"
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-
-int
-main (void)
-{
-return main ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"
-then :
-  ac_cv_lib_stdcpp_main=yes
-else $as_nop
-  ac_cv_lib_stdcpp_main=no
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.beam \
-    conftest$ac_exeext conftest.$ac_ext
-LIBS=$ac_check_lib_save_LIBS
-fi
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_stdcpp_main" >&5
-printf "%s\n" "$ac_cv_lib_stdcpp_main" >&6; }
-if test "x$ac_cv_lib_stdcpp_main" = xyes
-then :
-  printf "%s\n" "#define HAVE_LIBSTDC__ 1" >>confdefs.h
-
-  LIBS="-lstdc++ $LIBS"
-
-fi
 
 
 # These can sometimes be in the standard libraries
